scene/gui/scroll_container.cpp

```diff
@@ -426,36 +423,32 @@ void ScrollContainer::update_scrollbars() {
 	Ref<StyleBox> sb = get_theme_stylebox(SNAME("bg"));
 	size -= sb->get_minimum_size();
 
+	Size2 min = child_max_size;
+	h_scroll->set_visible(min.width > size.width);
+	v_scroll->set_visible(min.height > size.height);
+
 	Size2 hmin;
 	Size2 vmin;
-	if (horizontal_scroll_mode != SCROLL_MODE_DISABLED) {
+	if (h_scroll->is_visible_in_tree() && h_scroll->get_parent() == this) {
 		hmin = h_scroll->get_combined_minimum_size();
 	}
-	if (vertical_scroll_mode != SCROLL_MODE_DISABLED) {
+	if (v_scroll->is_visible_in_tree() && v_scroll->get_parent() == this) {
 		vmin = v_scroll->get_combined_minimum_size();
 	}
 
-	Size2 min = child_max_size;
-
-	bool hide_scroll_h = horizontal_scroll_mode != SCROLL_MODE_SHOW_ALWAYS && (horizontal_scroll_mode == SCROLL_MODE_DISABLED || horizontal_scroll_mode == SCROLL_MODE_SHOW_NEVER || (horizontal_scroll_mode == SCROLL_MODE_AUTO && min.width <= size.width));
-	bool hide_scroll_v = vertical_scroll_mode != SCROLL_MODE_SHOW_ALWAYS && (vertical_scroll_mode == SCROLL_MODE_DISABLED || vertical_scroll_mode == SCROLL_MODE_SHOW_NEVER || (vertical_scroll_mode == SCROLL_MODE_AUTO && min.height <= size.height));
-
 	h_scroll->set_max(min.width);
-	h_scroll->set_page(size.width - (hide_scroll_v ? 0 : vmin.width));
-	h_scroll->set_visible(!hide_scroll_h);
+	h_scroll->set_page(size.width - vmin.width);
 
 	v_scroll->set_max(min.height);
-	v_scroll->set_page(size.height - (hide_scroll_h ? 0 : hmin.height));
-	v_scroll->set_visible(!hide_scroll_v);
+	v_scroll->set_page(size.height - hmin.height);
 
 	// Avoid scrollbar overlapping.
-	h_scroll->set_anchor_and_offset(SIDE_RIGHT, ANCHOR_END, hide_scroll_v ? 0 : -vmin.width);
-	v_scroll->set_anchor_and_offset(SIDE_BOTTOM, ANCHOR_END, hide_scroll_h ? 0 : -hmin.height);
+	h_scroll->set_anchor_and_offset(SIDE_RIGHT, ANCHOR_END, -vmin.width);
+	v_scroll->set_anchor_and_offset(SIDE_BOTTOM, ANCHOR_END, -hmin.height);
 }
 
 void ScrollContainer::_scroll_moved(float) {
 	queue_sort();
-	update();
```

Comment:
For `ScrollContainer`, `queue_sort()` will eventually cause `update()` to be called, and all information can be confirmed at once, so there is no need to call `update()`.